MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -16554,18 +16551,14 @@ void ApplicationWindow::CatalogLogin() {
 }
 
 void ApplicationWindow::CatalogSearch() {
-  if (catalogSearch == NULL || catalogSearch) {
```

Comment:
@quantumsteve 
I don't think I actually wrote this code. I think it's likely that my name is there in the git blame due to a clang format. Regardless, given the nature of the check I think a smart pointer should work.